### ISSUE TITLE
Behavior removal bug in `MultiPointerScaleBehavior.detach()`

### ISSUE THREAD

**Issue by emtKazys** [BabylonJS/Babylon.js#18537]
`MultiPointerScaleBehavior` fails to remove itself correctly when `Node.removeBehavior()` is called, possibly event removing another, unrelated behaviour. Later this raises an exception since the addition of `this._ownerNode = null!;` in `MultiPointerScaleBehavior.detach()`.

This relates directly to this [forum post](https://forum.babylonjs.com/t/removing-a-behavior-inside-another-behaviors-detach-breaks-removebehavior/63549)

**Repro**

This playground sample adds the `MultiPointerScaleBehavior`, prints the number and names of the added behaviours, removes the `MultiPointerScaleBehavior` and does the same again. This should result in no behaviours left, but leaves one active.

- Bug repro on [playground](https://playground.babylonjs.com): https://playground.babylonjs.com/#M9DMC0 (vibe coded example)
- Expected result: "After removeBehavior() - 0 behavior(s): []"
- Current result: "After removeBehavior() - 1 behafior(s): ["MultiPointerScale"]"

**Desktop:**
 - OS: Linux (all)
 - Browser: Edge (all)
 - Version: 9.10.1

**Additional context**

The issue is that `Node.removeBehavior()` follows this sequence of events (assuming `MultiPointerScaleBehavior` is the only registered behaviour for illustrative purposes):

1. Get the index of the `MultiPointerScaleBehavior` => 2
2. Call `MultiPointerScaleBehavior.detach()`
3. Call `Node.removeBehavior()` on the two child behaviours
    - Removes behaviours at indices {0, 1}
4. Tries to remove the `MultiPointerScaleBehavior` from `Node._behaviors` using the index from step 1 (2).

The index used at step 4 is now stale and will either do nothing (best case) or remove a different behaviour.

Possible fix: effect step 4 before step 3 -

```ts
    public removeBehavior(behavior: Behavior<Node>): Node {
        const index = this._behaviors.indexOf(behavior);

        if (index === -1) {
            return this;
        }

        // These two lines have been swapped
        this._behaviors.splice(index, 1);
        this._behaviors[index].detach();

        return this;
    }
```

**Comment by sebavan** [comment]
cc @CedricGuillemet to see if he can have a look
